lib/monitor/stats.go

```diff
@@ -337,129 +352,58 @@ func (t *value) Collect(measure string, df DataFilter) {
-	return fmt.Sprintf("%f, %f, %f, %f", t.Min(), t.Max(), t.Avg(), t.Dev())
-}
-
-// Measurement represents the precise measurement of a specific thing to measure
-// example: I want to measure the time it takes to verify a signature, the
-// measurement "verify" will hold a wallclock Value, cpu_user Value, cpu_system
-// Value. A measurement is frequently updated with Measure given by the client.
-type Measurement struct {
-	Name   string
-	Wall   *value
-	User   *value
-	System *value
-	Filter DataFilter
-}
-
-// NewMeasurement returns a new measurements with this name
-func NewMeasurement(name string, df DataFilter) *Measurement {
-	return &Measurement{
-		Name:   name,
-		Wall:   newValue(),
-		User:   newValue(),
-		System: newValue(),
-		Filter: df,
-	}
-}
-
-// WriteHeader will write the header to the specified writer
-func (m *Measurement) WriteHeader(w io.Writer) {
-	fmt.Fprintf(w, "%s, %s, %s", m.Wall.Header(m.Name+"_wall"),
-		m.User.Header(m.Name+"_user"), m.System.Header(m.Name+"_system"))
-}
-
-// WriteValues will write a new entry for this entry in the writer
-// First compute the values then write to writer
-func (m *Measurement) WriteValues(w io.Writer) {
-	fmt.Fprintf(w, "%s, %s, %s", m.Wall.String(), m.User.String(), m.System.String())
-}
-
-// Update takes a measure received from the network and update the wall system
-// and user values
-func (m *Measurement) Update(measure Measure) {
-	dbg.Lvl2("Got measurement for", m.Name, measure.WallTime, measure.CPUTimeUser, measure.CPUTimeSys)
-	m.Wall.Store(measure.WallTime)
-	m.User.Store(measure.CPUTimeUser)
-	m.System.Store(measure.CPUTimeSys)
-}
-
-// Collect will call Collect on Wall- User- and System-time
-func (m *Measurement) Collect() {
-	m.Wall.Collect(m.Name, m.Filter)
-	m.User.Collect(m.Name, m.Filter)
-	m.System.Collect(m.Name, m.Filter)
-}
-
-// AverageMeasurements takes an slice of measurements and make the average
-// between them. i.e. it takes the average of the Wall value from each
-// measurements, etc.
-func AverageMeasurements(measurements []Measurement) Measurement {
-	m := NewMeasurement(measurements[0].Name, measurements[0].Filter)
-	walls := make([]*value, len(measurements))
-	users := make([]*value, len(measurements))
-	systems := make([]*value, len(measurements))
-	for i, m2 := range measurements {
-		m2.Collect()
-		walls[i] = m2.Wall
-		users[i] = m2.User
-		systems[i] = m2.System
-	}
-	m.Wall = AverageValue(walls...)
-	m.User = AverageValue(users...)
-	m.System = AverageValue(systems...)
-	return *m
+func (t *Value) HeaderFields() []string {
+	return []string{t.name + "_min", t.name + "_max", t.name + "_avg", t.name + "_dev"}
 }
 
-// String shows one measurement
-func (m *Measurement) String() string {
-	return fmt.Sprintf("{Measurement %s: wall = %v, system = %v, user = %v}", m.Name, m.Wall, m.User, m.System)
+// String returns the min, max, avg and dev of a Value
+func (t *Value) Values() []string {
+	return []string{fmt.Sprintf("%f", t.store)}
```

Comment:
I tried merging with current development, and stumbled upon this line:
Are you sure this is correct? `%f` assumes a floating value and `t.store` is an array of `float64`